### PR TITLE
Regex used to parse "git diff --staged" updated to allow spaces in fi…

### PR DESCRIPTION
**Pull request by tinamthomas**
Regex used to parse "git diff --staged" updated to allow spaces in filename

### CODE REVIEW

**Comment by dcRUSTy** [comment]
I remember trying something similiar before i gave up with regex 😅 

it does solves a lot of paths with spaces....
but i think breaks if a path contains " b/" 
eg PATH- `/plans/plan b/secretPlan.txt`

but it does solves a lot of paths with spaces. 👍 

Screenshot 2021-05-16 at 7 44 07 AM
Screenshot 2021-05-16 at 7 41 59 AM

**Comment by svishwanath-tw** [comment]
@dcRUSTy , @tinamthomas good research on your part. I'd have liked it better if the regex engine were set to 'Golang' instead of 'Javascript'. 
While I understand that regexes are fairly standard across programming languages, it is best to experiment with the one in the language of implementation.

**Comment by tinamthomas** [comment]
@dcRUSTy Thanks for calling that out 👍🏽

**Comment by tinamthomas** [comment]
Some flavours of regex would let you do this: ^diff --git a\/(.*) b\/(\1), unfortunately this won't work with golang 

Screenshot 2021-05-20 at 5 09 41 PM

**Comment by tinamthomas** [comment]
@dcRUSTy I've added a test that should cover the scenario you mentioned where the file name contains "b/" in the name
This is how the git diff --staged line would look for that fyi: 
diff --git a/folder b/blah.pem b/folder b/blah.pem

File: gitrepo/gitrepo.go [hunk 1]
```diff
@@ -51,7 +51,7 @@ func (repo GitRepo) GetDiffForStagedFiles() []Addition {
 
 	// Standard git diff header pattern
 	// ref: https://git-scm.com/docs/diff-format#_generating_patches_with_p
-	headerRegex := regexp.MustCompile("^diff --git a/([^ ]+) ")
+	headerRegex := regexp.MustCompile("^diff --git a/(.*) b/(.*)")
```
[the remark this answers is not in the document]
Comment (tinamthomas) [reply gitrepo/gitrepo.go]: yeah, they are meant to reference the same pattern. I've added a check for that.

**Review by svishwanath-tw — APPROVED** [review]

**Review by dineshba — APPROVED** [review]

**Pull request merged**